src/lsm/forest_fuzz.zig

```diff
@@ -0,0 +1,390 @@
+                    groove.put(&account);
+                    model.put(account.id, account) catch unreachable;
+                },
+                .get => |get| {
+                    switch (get) {
+                        .id => |id| {
+                            const groove = &env.forest.grooves.accounts;
+                            const Groove = @TypeOf(groove.*);
+
+                            // Prefetch by id.
+                            const Getter = struct {
+                                finished: bool = false,
+                                prefetch_context: Groove.PrefetchContext = undefined,
+                                fn prefetch_callback(prefetch_context: *Groove.PrefetchContext) void {
+                                    const getter = @fieldParentPtr(@This(), "prefetch_context", prefetch_context);
+                                    getter.finished = true;
+                                }
+                            };
+                            var getter = Getter{};
+                            groove.prefetch_setup(null);
+                            groove.prefetch_enqueue(id);
+                            groove.prefetch(Getter.prefetch_callback, &getter.prefetch_context);
+                            while (!getter.finished) env.tick();
+
+                            // Get by id.
+                            const lsm_account = groove.get(id);
+
+                            // Compare to model.
+                            const model_account = model.get(id);
+                            if (model_account == null) {
+                                assert(lsm_account == null);
+                            } else {
+                                assert(std.mem.eql(
+                                    u8,
+                                    std.mem.asBytes(&model_account.?),
+                                    std.mem.asBytes(lsm_account.?),
+                                ));
+                            }
+                        },
+                    }
+                },
+            }
+        }
+    }
+};
+
+fn random_int_exp(comptime Int: type, random: std.rand.Random, avg_int: Int) Int {
+    return @floatToInt(Int, @trunc(random.floatExp(f64) * @intToFloat(f64, avg_int)));
+}
+
+fn random_id(comptime Int: type, random: std.rand.Random) Int {
+    // We have two opposing desires for random ids:
+    const avg_int: Int = if (random.boolean())
+        // 1. We want to cause many collisions.
+        8
+    else
+        // 2. We want to generate enough ids that the cache can't hold them all.
+        Environment.cache_entries_max;
+    return random_int_exp(Int, random, avg_int);
+}
+
+pub fn fuzz(fuzz_ops: []const FuzzOp) !void {
+    // Init mocked storage.
+    var storage = try Storage.init(
+        allocator,
+        Environment.size_max +
+            // TODO this is totally arbitrary
+            16 * 1024 * 1024 * 1024,
```

Comment:
No need to add extra here; `size_max` includes the grid size.

Reply:
I could have sworn I was getting storage errors before adding this, but I cannot reproduce by removing it now. :shrug:

Reply:
Well, if it does need more storage, just increase the `size_max`. I only noticed because I don't have 16GB free to run the fuzz test :sweat_smile:.